### ISSUE TITLE
improvement : add domain name validation

### ISSUE THREAD

**Issue by neyzth** [fabian-hiller/valibot#894]
Hello, as a IT guy i have to work sometimes with domain name, not url. Something to check if the input like :
```txt
corebackbone.par.franceix.net
```
or
```txt
vl223.par-itx5-dist-2.cdn77.com
```
have a valid domain name syntax.

Is this possible to add a action for validating this kind of inputs please ?

**Comment by fabian-hiller** [comment]
Yes, we can add a `domain` action. Is there any regex you would recommend? Would you be interested in creating a PR?

**Comment by neyzth** [comment]
hi, this kind of regex i guess :
```REGEX
^((?!-)[A-Za-z0-9-]{1,63}(?<!-).)+[A-Za-z]{2,}$
```

**Comment by fabian-hiller** [comment]
Are you interested in creating a PR that implements the actions and adds tests? This action is very similar to other actions like `email` or `decimal`. You could simply copy one of them and use it as a template.

**Comment by JacKyDev** [comment]
If the questioner doesn’t want to, I’d do it. However, I’d only be able to manage it next week.

I wanted to dive deeper into the valibot code anyway. But only if you agree, as the questioner hasn’t responded yet.

**Comment by paksa92** [comment]
I implemented it: #907

**Comment by fabian-hiller** [comment]
I am focusing on our v1 release at the moment. In the meantime, I recommend using the regex with our `regex` action as a workaround.

**Comment by JacKyDev** [comment]
@fabian-hiller : A quick general question about PRs since I’m looking at one right now. Should the `.mdx` files for the website, like `menu.md` and `properties.ts`, also be updated? Otherwise, it would mean the features are there but not visible anywhere (like in the docs).

**Comment by fabian-hiller** [comment]
Yes, but I can also take care of the details because I know it can be hard to understand all the connections.

**Comment by tats-u** [comment]
![image](https://github.com/user-attachments/assets/ae55e05e-b5bc-482b-8440-380570126849)
https://makenowjust-labs.github.io/recheck/playground/

```js
/^((?!-)[A-Za-z0-9-]{1,63}(?<!-).)+[A-Za-z]{2,}$/.test('0' + '0-'.repeat(30) + '\x00')
```

took about 10s to generate `false`.

**Comment by paksa92** [comment]
@tats-u we agreed on this final regex pattern: `/^(?=.{1,253}$)(?!-)([a-z0-9-]{1,63}(?<!-)\.)+[a-z]{2,63}$/iu` in #907.

Which is safe according to the tool you provided:
<img width="1302" alt="image" src="https://github.com/user-attachments/assets/9aee8a55-6df6-4807-8589-72fc980a3301">

**Comment by tats-u** [comment]
I see. That's a good news.

**Comment by fabian-hiller** [comment]
Not yet but soon. I will review and merge #907 probably on the weekend. In the meantime feel free to use this as a workaround:

```ts
const DOMAIN_REGEX = /^(?!-)([a-z0-9-]{1,63}(?<!-)\.)+[a-z]{2,6}$/iu;
const DomainSchema = v.pipe(v.string(), v.regex(DOMAIN_REGEX));
```

**Comment by tats-u** [comment]
Shouldn't we reopen this issue?

**Comment by fabian-hiller** [comment]
We can. Would you be interested in creating a PR?

**Comment by yslpn** [comment]
I don't understand what goes wrong with the previous PR?

https://github.com/fabian-hiller/valibot/pull/907

**Comment by yslpn** [comment]
Hello. Help me understand what status this feature is in? It seems the issue is still relevant, it needs to be reopened.

cc @paksa92 @fabian-hiller @neyzth
